test/json_unittest.cc

```diff
@@ -16,12 +29,40 @@ void GuardJsonException(std::function<void()> test) {
 }
 
 
+// Type values are distinct.
+
+TEST(Distinct, Types) {
+  std::vector<json::Type> all_types({
+    json::NullType, json::BooleanType, json::NumberType, json::StringType,
+    json::ArrayType, json::ObjectType
+  });
+  std::set<json::Type> all_types_set(all_types.begin(), all_types.end());
```

Comment:
You'll want to `#include <set>` for this...

Reply:
Done.